src/test/java/se/sundsvall/ai/flow/TestDataFactory.java

```diff
@@ -0,0 +1,153 @@
+package se.sundsvall.ai.flow;
+
+import static se.sundsvall.ai.flow.model.flow.FlowInput.Cardinality.MULTIPLE_VALUED;
+import static se.sundsvall.ai.flow.model.flow.FlowInput.Cardinality.SINGLE_VALUED;
+import static se.sundsvall.ai.flow.model.flow.InputType.TEXT;
+
+import java.util.Base64;
+import java.util.List;
+import java.util.UUID;
+import se.sundsvall.ai.flow.api.model.RenderRequest;
+import se.sundsvall.ai.flow.model.Session;
+import se.sundsvall.ai.flow.model.flow.Flow;
+import se.sundsvall.ai.flow.model.flow.FlowInput;
+import se.sundsvall.ai.flow.model.flow.FlowInputRef;
+import se.sundsvall.ai.flow.model.flow.RedirectedOutput;
+import se.sundsvall.ai.flow.model.flow.Step;
+import se.sundsvall.ai.flow.service.flow.ExecutionState;
+import se.sundsvall.ai.flow.service.flow.StepExecution;
+
+public final class TestDataFactory {
+
+	public static Session createNewSession() {
+		var session = new Session().withFlow(createFlow());
+		session.getFlow().getInputs().forEach(flowInput -> session.addInput(flowInput.getId(), Base64.getEncoder().encodeToString("value".getBytes())));
+
+		return session;
+	}
+
+	public static Session createSessionWithStepExecutions() {
+		var session = new Session().withFlow(createFlow());
+		session.getFlow().getInputs().forEach(flowInput -> session.addInput(flowInput.getId(), Base64.getEncoder().encodeToString("value".getBytes())));
+		session.addStepExecution("step", createStepExecution());
+
+		return session;
+	}
+
+	public static Flow createFlow() {
+		var flow = new Flow();
+		flow.setName("Tjänsteskrivelse");
+		flow.setDescription("Ett Intric AI-flöde för tjänsteskrivelser");
+		flow.setId("tjansteskrivelse");
+		flow.setInputPrefix("#####");
+		flow.setDefaultTemplateId("ai-mvp.tjansteskrivelse");
+		flow.setInputs(List.of(
+			createFlowInput1(),
+			createFlowInput2(),
+			createFlowInput3(),
+			createFlowInput4()));
+		flow.setSteps(List.of(
+			createStep1(),
+			createStep2(),
+			createStep3()));
+		return flow;
+	}
+
+	public static StepExecution createStepExecution() {
+		var stepExecution = new StepExecution(UUID.randomUUID(), createStep1(), List.of());
+		stepExecution.setState(ExecutionState.PENDING);
+		stepExecution.setOutput("output");
+
+		return stepExecution;
+	}
+
+	public static FlowInput createFlowInput1() {
+		var flowInput = new FlowInput();
+		flowInput.setName("Ärendenummer");
+		flowInput.setId("arendenummer");
+		flowInput.setType(TEXT);
+		flowInput.setCardinality(SINGLE_VALUED);
+		flowInput.setPassthrough(true);
+		return flowInput;
+	}
+
+	public static FlowInput createFlowInput2() {
```

Comment:
Could possibly have some better naming for these methods? Like instead of 2 use name? idk.